src/main/java/org/jabref/gui/entryeditor/CommentsTab.java

```diff
@@ -139,17 +138,17 @@ protected void setupPanel(BibEntry entry, boolean compressed) {
         // Show "Hide" button only if user-specific comment field is empty. Otherwise, it is a strange UI, because the
         // button would just disappear and no change **in the current** editor would be made
         if (entryEditorPreferences.shouldShowUserCommentsFields() && !entry.hasField(userSpecificCommentField)) {
-            Button hideDefaultOwnerCommentButton = new Button(Localization.lang("Hide user comments"));
-            hideDefaultOwnerCommentButton.setOnAction(e -> {
-                var labelForField = gridPane.getChildren().stream().filter(s -> s instanceof FieldNameLabel).filter(x -> ((FieldNameLabel) x).getText().equals(userSpecificCommentField.getDisplayName())).findFirst();
-                labelForField.ifPresent(label -> gridPane.getChildren().remove(label));
-                fieldEditorForUserDefinedComment.ifPresent(f -> gridPane.getChildren().remove(f.getNode()));
-                editors.remove(userSpecificCommentField);
-
-                entryEditorPreferences.setShowUserCommentsFields(false);
-                setupPanel(entry, false);
+            Button toggleCommentsButton = new Button("Hide user comments");
```

Comment:
You might want to use a [toggleButton](https://jenkov.com/tutorials/javafx/togglebutton.html) here since there are two possible states for the button: Show or Hide comments.